core/src/env_platform/utility.bal

```diff
@@ -117,3 +117,89 @@ function getApplicationCode(string applicationType) returns string|error {
+function getCurrentDateObject() returns Date {
+    time:Time time = time:currentTime();
+    return {
+        "year": time:getYear(time),
+        "month": time:getMonth(time),
+        "day": time:getDay(time),
+        "hour": time:getHour(time),
+        "minute": time:getMinute(time)
+    };
+}
+
+# The `constructAssignment` function will construct the assignment from the given AssignedMinistry record.
+# 
+# + assignedMinistry - AssignedMinistry record.
+# + return - This function will return a assignment json or an error if the Data record is not converted to the json, 
+# Mongodb:DatabaseError or prerequisite ministry is not found.
+function constructAssignment(AssignedMinistry assignedMinistry) returns json|error {
+
+    json data;
+    Ministry? prerequisite = assignedMinistry?.prerequisite;
+
+    // If prerequisite is available.
+    if (prerequisite is Ministry) {
+
+        // Check the prerequisite ministry availablity.
+        if (check isMinistry(prerequisite.id)) {
+            data = {
+                id: assignedMinistry.ministry.id,
+                name: assignedMinistry.ministry.name,
+                prerequisiteId: prerequisite.id,
+                prerequisiteName: prerequisite.name,
+                status: [
+                        {
+                            progress: "New",
+                            timestamp: check json.constructFrom(getCurrentDateObject())
+                        }
+                    ]
+            };
+        } else {
+            return error("Invalid Operation", message = "There is no prerequisite ministry found with the ID: " + prerequisite.id + ".");
+        }
+    } else {
+        data = {
+            id: assignedMinistry.ministry.id,
+            name: assignedMinistry.ministry.name,
+            status: [
+                    {
+                        progress: "New",
+                        timestamp: check json.constructFrom(getCurrentDateObject())
+                    }
+                ]
+        };
+    }
+    return data;
+}
+
+# The `constructAssignmentArray` function will construct the assignments array by pushing the new assigned ministry.
+# 
+# + assignedMinistry - AssignedMinistry record which must be add to the assignments.
+# + assignments - All the assignments of an application.
+# + return - This function will return either constructed assigned ministry array or an error.
+function constructAssignmentArray(AssignedMinistry assignedMinistry, json[] assignments) returns json|error {
+
+    error duplicateError = error("Ministry Already Assigned", message = "The ministry is with ministry ID: " + assignedMinistry.ministry.id + " is already assigned");
+    boolean isError = false;
+
+    // Check if the ministry is already assigned.
+    assignments.forEach(function (json assignment) {
+        map<json> assignmentMap = <map<json>>assignment;
+        if (assignmentMap.id == assignedMinistry.ministry.id) {
+            isError = true;
+            return;
```

Comment:
```suggestion
            break;
```
This should break the loop, I'm not sure if **return** would do the same

Reply:
The first thing which we have to check is whether we can use break there. If possible then we should break else we'll need to think.

Reply:
<img width="624" alt="Screenshot 2020-05-20 at 18 07 32" src="https://user-images.githubusercontent.com/32588172/82446743-da013080-9ac4-11ea-8237-ee3975c32f16.png">
The break wont work for this. If we want to exit from the function we have to use return.

Reply:
@sanjulamadurapperuma @deshankoswatte return will exit from the forEach loop.

Reply:
<img width="595" alt="Screenshot 2020-05-20 at 18 12 14" src="https://user-images.githubusercontent.com/32588172/82447125-71668380-9ac5-11ea-845f-cdcbca5edf46.png">

Reply:
> <img alt="Screenshot 2020-05-20 at 18 07 32" width="624" src="https://user-images.githubusercontent.com/32588172/82446743-da013080-9ac4-11ea-8237-ee3975c32f16.png">
> 
> The break wont work for this. If we want to exit from the function we have to use return.

What's the error message? I've seen break been used elsewhere to exit loops in other projects

Reply:
@sanjulamadurapperuma Please refer the above.

Reply:
@yashodgayashan Hmmm...strange error message, but if **return** works for this scenario then leave as is.

Reply:
@sanjulamadurapperuma @deshankoswatte return is either break it. I will ask from ballerina community.

Reply:
@yashodgayashan so you mean return doesn't exit the loop after all?

Reply:
@sanjulamadurapperuma I just check with some prints it will execute even after the return since it just execute from the function but forEach loop continue executing. I asked it from ballerina community. I will let you know once I got an answer.

Reply:
That shouldn't happen...yeah we'll wait for a answer from the community

Reply:
Or else we can use the normal forEach loop. But there should be an answer.

Reply:
@sanjulamadurapperuma @deshankoswatte There is no way to terminate this. They suggest normal forEach loop. I will update the code.

Reply:
Yeah @yashodgayashan will continue with the normal for loop.

Reply:
@yashodgayashan yes seems like we have to do that, 

> > <img alt="Screenshot 2020-05-20 at 18 07 32" width="624" src="https://user-images.githubusercontent.com/32588172/82446743-da013080-9ac4-11ea-8237-ee3975c32f16.png">
> > The break wont work for this. If we want to exit from the function we have to use return.
> 
> What's the error message? I've seen break been used elsewhere to exit loops in Ballerina in other projects

Regarding this comment, I checked on this and it seems like **break** works for while loop

Reply:
<img width="604" alt="Screenshot 2020-05-20 at 18 53 42" src="https://user-images.githubusercontent.com/32588172/82451154-3c5d2f80-9acb-11ea-8edd-24aebad9f62c.png">
I just updated the code with normal foreach loop.